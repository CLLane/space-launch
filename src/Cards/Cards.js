import React from 'react';

export const Cards = ({ launch }) => {
 
  let newName = launch.name.split('|').map(word => word)
  return (
  <article>
  <h1>{newName[0]}</h1>
  <h1>{newName[1] || null}</h1>
  <h1>{newName[2] || null}</h1>
  {/* <h1>{newName[3] || null}</h1>
  <h1>{newName[4] || null}</h1> */}
    <p>{launch.windowstart}</p>
    <a href={launch.vidURLs}>Launch Video</a>
    <div>{launch.location.pads.map(pad =>
      <div>
        <p>{pad.name}</p>
        <a href={pad.mapURL}>Launch Pad Location</a>
        <p>{pad.latitude}</p>
        <p>{pad.longitude}</p>
        <div>{pad.agencies && pad.agencies.map(agency =>
          <div>
            <a href={agency.wikiURL}>{agency.name}</a>
            <p>{agency.abbrev}</p>
            <p>{agency.countryCode}</p>
            <div>{agency.infoURLs.map(url =>
              <div>
                <a href={url} >Link</a>
              </div>
            )}</div>
          </div>
        )}</div>
      </div>
    )}</div>
    </article>
    )
}