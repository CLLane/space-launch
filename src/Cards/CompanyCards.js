import React from 'react';

export const CompanyCards = ({ company }) => {
  let abbrev = company.countryCode.split(',')
  return (
    <article>
      <h1>{company.name}</h1>
      <h3>Countries:</h3>
    <div className='country-container'>
    <div className='abbrev'>
        <p>{abbrev[0]}</p>
        <p>{abbrev[1]}</p>
        <p>{abbrev[2]}</p>
        <p>{abbrev[3]}</p>
      </div>
     <div className='abbrev'>
        <p>{abbrev[4]}</p>
        <p>{abbrev[5]}</p>
        <p>{abbrev[6]}</p>
        <p>{abbrev[7]}</p>
      </div>
     <div className='abbrev'>
        <p>{abbrev[8]}</p>
        <p>{abbrev[9]}</p>
        <p>{abbrev[10]}</p>
        <p>{abbrev[11]}</p>
      </div>
     <div className='abbrev'>
        <p>{abbrev[12]}</p>
        <p>{abbrev[13]}</p>
        <p>{abbrev[14]}</p>
        <p>{abbrev[15]}</p>
      </div>
    </div>
    <h3>Company:</h3>
    <p>{company.abbrev}</p>
      {company.infoURL && <a className='wiki-link' href={company.infoURL}>Info</a> || <p>Sorry No Info</p>}
      {company.wikiURL && <a className='info-link' href={company.wikiURL}>Wiki</a> || <p>Sorry No Wiki</p>}
    </article>
  )
}