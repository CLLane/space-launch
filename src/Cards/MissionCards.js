import React from 'react';
import PropTypes from "prop-types";

export const MissionCard = ({ mission }) => {
  return (
    <article>
      <h1>{mission.name}</h1>
      <p>{mission.description}</p>
      <a className='info-link'href={mission.infoURL}>Info</a>
      <a className='wiki-link'href={mission.wikiURL}>Wiki</a>
    </article>
  )
}

MissionCard.propTypes = {
  mission: PropTypes.object.isRequired
}
