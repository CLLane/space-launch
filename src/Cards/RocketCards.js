import React from 'react';

export const RocketCards = ({ rocket }) => {
  console.log(rocket)
  return (
    <article>
      <h1>{rocket.name}</h1>
      <p>Family:{rocket.family.name}</p>
      <a href={rocket.wikiURL}>Wiki</a>
      <a href={rocket.infoURL}>Info</a>
      <img className='rocket-images' alt={rocket.name} src={rocket.imageURL}></img>
    </article>
  )
}