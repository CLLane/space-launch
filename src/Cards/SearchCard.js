import React from 'react'
import ReactPlayer from "react-player";

export const SearchCard = ({ result }) => {

  let urlString;
  if (!result.vidURLs) {
    urlString = ''
  } else {
    urlString = result.vidURLs.toString()
  }
  return (
    <article>
      <h1>{result.name}</h1>
      {result.configuration && <p>Config:{result.configuration}</p>}
      {result.wikiURL && <a className='wiki-link' href={result.wikiURL}>Wiki</a>}
      {result.infoURLs[0] && <a className='info-link' href={result.infoURLs[0]}>Info</a>}
      {result.imageURL && <img className='rocket-images' alt='' src={result.imageURL}></img>}
      {result.windowstart && <p>{result.windowstart}</p>}
      {urlString && <ReactPlayer url={urlString} />}
      {result.failreason && <p>{result.failreason}</p>}
      {result.pads > 0 && <p>{result.pads[0].name}</p>}
      {result.countryCode && <p>{result.countryCode}</p>}
      {result.abbrev && <p>{result.abbrev}</p>}
      {result.description && <p>{result.description}</p>}
      
    </article>
  )
}