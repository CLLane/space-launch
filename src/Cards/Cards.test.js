import React from "react";
import { shallow } from "enzyme";
import Cards from "../Cards/Cards";
import { MissionCard } from "../Cards/MissionCards";
import { CompanyCards } from "./CompanyCards";
import { RocketCards } from "./RocketCards";
import { SearchCard } from "./SearchCard";


describe('Cards', () => {
  const mockLaunch = {
    name: 'This | is | a Name',
    windowstart: 'this is a date',
    vidURLs: '',
    location: {
      pads: [{
        name: 'launch site',
        agencies: [{
          abbrev: 'stff',
          countryCode: 'GMT'
        }]
      }]
    }
  }
  it('should match snapshot', () => {
    let wrapper = shallow(<Cards launch={mockLaunch} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('MissionCards', () => {
  let mockMission = {
    name: 'name',
    description: 'some launch things',
    infoURL: 'this is a link',
    wikiURL: 'also a link'
  }
  it('should match snapshot', () => {
    let wrapper = shallow(<MissionCard mission={mockMission} />)
    expect(wrapper).toMatchSnapshot();
  });
});

describe('CompanyCards', () => {
  let mockCompany = {
    name: 'A country',
    countryCode: 'a,b,c,d',
    abbrev: 'stff',
    infoURL: 'this is a link',
    wikiURL: 'also a link'
  };
  it('Should match snapshot', () => {
    let wrapper = shallow(<CompanyCards company={mockCompany} />);
    expect(wrapper).toMatchSnapshot();
  })
})

describe('RocketCards', () => {
  let mockRocket = {
    name: 'falcon',
    family: {
      name:'birds'
    },
    wikiURL: 'This is a link',
    infoURL: 'Also a link',
    imageURL: 'Image'
  };
  it('should match the snapshot', () => {
    let wrapper = shallow(<RocketCards rocket={mockRocket} />);
    expect(wrapper).toMatchSnapshot()

  })
})

describe('SearchCard', () => {
  let mockResult = {
    name: 'Chris',
    configuration: 'Hey',
    wikiURL: 'url',
    infoURLs: ['url'],
    imageURL: 'url',
    windowstart: 'date',
    vidURLs: '',
    failreason: 'blew up',
    pads: ['cheese'],
    countryCode: 'GMT',
    description: 'Snapshot Test'
  };
  it('should match the snapshot', () => {
    let wrapper = shallow(<SearchCard result={mockResult} />)
    expect(wrapper).toMatchSnapshot()
  })
})