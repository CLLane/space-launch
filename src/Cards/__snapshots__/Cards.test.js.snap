// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards should match snapshot 1`] = `
<article>
  <h1>
    This 
  </h1>
  <h1>
     is 
  </h1>
  <h1>
     a Name
  </h1>
  <p>
    this is a date
  </p>
  <div>
    <div>
      <h3>
        Launch Pad:
      </h3>
      <p>
        launch site
      </p>
      <div>
        <div>
          <h3>
            Pad Company:
          </h3>
          <p>
            stff
          </p>
          <h3>
            Pad Country:
          </h3>
          <p>
            GMT
          </p>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`CompanyCards Should match snapshot 1`] = `
<article>
  <h1>
    A country
  </h1>
  <h3>
    Countries:
  </h3>
  <div
    className="country-container"
  >
    <div
      className="abbrev"
    >
      <p>
        a
      </p>
      <p>
        b
      </p>
      <p>
        c
      </p>
      <p>
        d
      </p>
    </div>
    <div
      className="abbrev"
    >
      <p />
      <p />
      <p />
      <p />
    </div>
    <div
      className="abbrev"
    >
      <p />
      <p />
      <p />
      <p />
    </div>
    <div
      className="abbrev"
    >
      <p />
      <p />
      <p />
      <p />
    </div>
  </div>
  <h3>
    Company:
  </h3>
  <p>
    stff
  </p>
  <a
    className="wiki-link"
    href="this is a link"
  >
    Info
  </a>
  <a
    className="info-link"
    href="also a link"
  >
    Wiki
  </a>
</article>
`;

exports[`MissionCards should match snapshot 1`] = `
<article>
  <h1>
    name
  </h1>
  <p>
    some launch things
  </p>
  <a
    className="info-link"
    href="this is a link"
  >
    Info
  </a>
  <a
    className="wiki-link"
    href="also a link"
  >
    Wiki
  </a>
</article>
`;

exports[`RocketCards should match the snapshot 1`] = `
<article>
  <h1>
    falcon
  </h1>
  <p>
    Family:
    birds
  </p>
  <a
    className="wiki-link"
    href="This is a link"
  >
    Wiki
  </a>
  <a
    className="info-link"
    href="Also a link"
  >
    Info
  </a>
  <img
    alt="falcon"
    className="rocket-images"
    src="Image"
  />
</article>
`;

exports[`SearchCard should match the snapshot 1`] = `
<article>
  <h1>
    Chris
  </h1>
  <p>
    Config:
    Hey
  </p>
  <a
    className="wiki-link"
    href="url"
  >
    Wiki
  </a>
  <a
    className="info-link"
    href="url"
  >
    Info
  </a>
  <img
    alt=""
    className="rocket-images"
    src="url"
  />
  <p>
    date
  </p>
  <p>
    blew up
  </p>
  <p>
    GMT
  </p>
  <p>
    Snapshot Test
  </p>
</article>
`;
